app/gilded-rose.ts

```diff
@@ -18,52 +18,42 @@ export class GildedRose {
   }
 
   updateQuality() {
+    // Quality out of bounds
+    let increment = 1;
     for (let i = 0; i < this.items.length; i++) {
-      if (this.items[i].name != 'Aged Brie' && this.items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-        if (this.items[i].quality > 0) {
-          if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
-            this.items[i].quality = this.items[i].quality - 1
-          }
-        }
-      } else {
-        if (this.items[i].quality < 50) {
-          this.items[i].quality = this.items[i].quality + 1
-          if (this.items[i].name == 'Backstage passes to a TAFKAL80ETC concert') {
-            if (this.items[i].sellIn < 11) {
-              if (this.items[i].quality < 50) {
-                this.items[i].quality = this.items[i].quality + 1
-              }
-            }
-            if (this.items[i].sellIn < 6) {
-              if (this.items[i].quality < 50) {
-                this.items[i].quality = this.items[i].quality + 1
-              }
-            }
-          }
-        }
+      if (this.items[i].name == 'Sulfuras, Hand of Ragnaros') {
+        continue;
       }
-      if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
-        this.items[i].sellIn = this.items[i].sellIn - 1;
-      }
-      if (this.items[i].sellIn < 0) {
-        if (this.items[i].name != 'Aged Brie') {
-          if (this.items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-            if (this.items[i].quality > 0) {
-              if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
-                this.items[i].quality = this.items[i].quality - 1
-              }
-            }
+      else {
+        if (this.items[i].name == 'Backstage passes to a TAFKAL80ETC concert') {
+          if (this.items[i].sellIn < 0) {
+            this.items[i].quality = 0;
+            continue;
           } else {
-            this.items[i].quality = this.items[i].quality - this.items[i].quality
-          }
-        } else {
-          if (this.items[i].quality < 50) {
-            this.items[i].quality = this.items[i].quality + 1
+            increment = 3 - Math.floor(this.items[i].sellIn / 5);
           }
         }
+        else if (this.items[i].name == 'Aged Brie') {
+          increment = 1;
+        }
+        else {
+          increment = -1;
+        }
+        if (this.items[i].sellIn < 0) {
+          increment *= 2;
+        }
+        if (this.items[i].name.indexOf('Conjured') > -1) {
+          increment *= 2;
+        }
+        if (increment > 0) {
+          this.items[i].quality = Math.min(this.items[i].quality + increment, 50);
+        }
+        else {
+          this.items[i].quality = Math.max(this.items[i].quality + increment, 0);
+        }
+        this.items[i].sellIn--;
```

Comment:
I think this is a lot cleaner than the original - but there's a bit more you can do in terms of commonising actions and separating out the different cases - they're still quite tied up together and not the easiest to read.